src/__internal__/fieldset/fieldset.component.js

```diff
@@ -79,12 +80,18 @@ Fieldset.propTypes = {
   styleOverride: PropTypes.shape({
     root: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
     legend: PropTypes.oneOfType([PropTypes.func, PropTypes.object])
-  })
+  }),
+  /** Flag to configure component as mandatory HTML 5 */
+  required: PropTypes.bool,
+  /** Flag to configure component as mandatory */
+  'aria-required': PropTypes.bool
 };
 
 Fieldset.defaultProps = {
   inline: false,
-  styleOverride: {}
+  styleOverride: {},
+  required: false,
```

Comment:
This could result in a component having `aria-required={false} require={true}` which is incorrect. We need the props to be optional.

e.g.

```jsx
<input
{...(required && {required)}
/>
```

This way if the prop is false or undefined it won't be rendered to the DOM. This will resolve most of the snapshots below too.

Reply:
Done